app/services/customers/create_service.rb

```diff
@@ -78,47 +79,56 @@ def create(**args)
 
     private
 
-    # NOTE: Check if a payment provider is configured in the organization and
-    #       force creation of provider customers
-    def create_billing_configuration(customer, billing_configuration = {})
-      create_stripe_customer = customer.organization.stripe_payment_provider&.create_customers
-      create_stripe_customer ||= (billing_configuration || {})[:provider_customer_id]
-      return unless create_stripe_customer
+    def create_billing_configuration(customer, billing_configuration = {}, is_api:)
+      return if billing_configuration.blank?
 
-      customer.update!(payment_provider: 'stripe')
+      create_provider_customer = billing_configuration[:sync_with_provider]
+      create_provider_customer ||= billing_configuration[:provider_customer_id]
+      return unless create_provider_customer
 
-      create_result = PaymentProviderCustomers::CreateService.new(customer).create_or_update(
-        customer_class: PaymentProviderCustomers::StripeCustomer,
-        payment_provider_id: customer.organization.stripe_payment_provider&.id,
-        params: billing_configuration,
-      )
-      create_result.throw_error unless create_result.success?
+      customer.update!(payment_provider: billing_configuration[:payment_provider]) if is_api
+
+      create_or_update_provider_customer(customer, billing_configuration)
     end
 
     def handle_api_billing_configuration(customer, params, new_customer)
-      unless params.key?(:billing_configuration)
-        create_billing_configuration(customer) if new_customer
-        return
-      end
+      return unless params.key?(:billing_configuration)
```

Comment:
With this change we will not create the stripe customer automatically if the organization option is checked. Is it intended?

Reply:
exactly, we only check property on the customer level